packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.java

```diff
@@ -29,7 +42,152 @@ public void onCreate(ModuleRegistry moduleRegistry) {
     mModuleRegistry = moduleRegistry;
   }
 
+  @Override
```

Comment:
IMHO, it is almost always a good idea to try follow Don't repeat yourself principle. Huge chunks on code in here are identical to parts of UpdatedController. It might be either exposed by controller and used here, or moved to some other class and used by both module and controller. We benefit from that by not forcing ourselves to remember that bugs need to be fixed in two places, if spotted.